internal/schema/resourcedata.go

```diff
@@ -31,16 +35,42 @@ func GetResourceDataChangedKeys(keys []string, d resourceDataProvider) map[strin
 	return changed
 }
 
+// GetResourceDataListElementChanges returns list element changes
 func GetResourceDataListElementChanges(keys []string, listKeyName string, listIndex int, d resourceDataProvider) map[string]interface{} {
 	changed := make(map[string]interface{})
 	if !d.HasChange(listKeyName) {
 		return changed
 	}
-	old, new := d.GetChange(listKeyName)
+	old, newName := d.GetChange(listKeyName)
 	oldList := old.([]interface{})
-	newList := new.([]interface{})
+	newList := newName.([]interface{})
 	if len(oldList) < listIndex || len(newList) < listIndex {
 		return changed
 	}
+	if len(oldList) == 0 {
+		return newList[0].(map[string]interface{})
+	}
 	return getMapChangedKeys(keys, oldList[listIndex].(map[string]interface{}), newList[listIndex].(map[string]interface{}))
 }
+
+// IsDataElementAdded - checks if a data element added
+func IsDataElementAdded(_ []string, listKeyName string, d resourceDataProvider) bool {
```

Comment:
Let's merge IsDataElementAdded and IsDataElementRemoved functions to make it more clear